python/tank/descriptor/io_descriptor/downloadable.py

```diff
@@ -57,53 +60,122 @@ def download_local(self):
+            raise TankDescriptorIOError("Failed to download into path %s: %s" % (temporary_path, e))
 
         log.debug("Attempting to move descriptor %s from temporary path %s to target path %s." % (
             self, temporary_path, target)
         )
+
+        move_succeeded = False
+
         try:
             # atomically rename the directory temporary_path to the target.
+            # note: this is so that we don't end up with a partial payload in the target
+            # location. All or nothing.
             os.rename(temporary_path, target)
+            move_succeeded = True
             log.debug("Successfully moved the downloaded descriptor to target path: %s." % target)
+
         except Exception as e:
-            # if the target path does not already exist, something else might have gone wrong.
+
+            # if the target path already exists, this means someone else is either
+            # moving things right now or have moved it already, so we are ok.
             if not os.path.exists(target):
-                log.error("Failed to move descriptor from the temporary path %s " % temporary_path +
-                          " to the bundle cache %s: %s" % (target, e))
-                raise TankError("Failed to move descriptor from the temporary path %s " % temporary_path +
-                                " to the bundle cache %s: %s" % (target, e))
-        else:
-            self._post_download(target)
-        finally:
-            if os.path.exists(temporary_path):
-                log.debug("Removing temporary path: %s" % temporary_path)
+                # the target path does not exist. so the rename failed for other reasons.
+
+                # if the rename did not work, it may be because the files are locked and
+                # cannot be deleted. This can for example happen if an antivirus software
+                # (on windows) has been triggered because of the download and has locked on
+                # to the files. In this case, we try to copy the files and then remove the
+                # temp payload - this is slower, but safer, and we can gracefully fail and
+                # continue in case the deletion fails.
+                log.warning(
+                    "Failed to move descriptor %s from the temporary path %s "
+                    "to the bundle cache %s. Will attempt to copy it instead. "
+                    "Error: %s" % (self, temporary_path, target, e)
+                )
+
+                try:
+                    # copy first then delete all files in target.
+                    # if deletion fails this will log and gracefully continue.
+                    log.debug(
+                        "Performing 'copy then delete' style move on %s -> %s" % (
+                            temporary_path,
+                            target
+                        )
+                    )
+                    filesystem.move_folder(temporary_path, target)
```

Comment:
Maybe I am missing something, but this is where I was expecting the `system-metadata` folder to be created and after the copy the download_completed file would be touched. A failed download to the temporary folder doesn't need to be tracked, but a failed copy o the individual files to the final location needs to.